kurl_util/cmd/pvmigrate/main.go

```diff
@@ -1,15 +1,133 @@
 package main
 
 import (
+	"context"
+	"flag"
 	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
 
+	clusterspace "github.com/replicatedhq/kurl/pkg/cluster/space"
 	"github.com/replicatedhq/kurl/pkg/version"
 	"github.com/replicatedhq/pvmigrate/pkg/migrate"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
 func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
 	fmt.Printf("Running pvmigrate build:\n")
 	version.Print()
 
-	migrate.Cli()
+	var skipFreeSpaceCheck bool
+	var opts migrate.Options
+	flag.StringVar(&opts.SourceSCName, "source-sc", "", "storage provider name to migrate from")
+	flag.StringVar(&opts.DestSCName, "dest-sc", "", "storage provider name to migrate to")
+	flag.StringVar(&opts.RsyncImage, "rsync-image", "eeacms/rsync:2.3", "the image to use to copy PVCs - must have 'rsync' on the path")
+	flag.StringVar(&opts.Namespace, "namespace", "", "only migrate PVCs within this namespace")
+	flag.BoolVar(&opts.SetDefaults, "set-defaults", false, "change default storage class from source to dest")
+	flag.BoolVar(&opts.VerboseCopy, "verbose-copy", false, "show output from the rsync command used to copy data between PVCs")
+	flag.BoolVar(&opts.SkipSourceValidation, "skip-source-validation", false, "migrate from PVCs using a particular StorageClass name, even if that StorageClass does not exist")
+	flag.BoolVar(&skipFreeSpaceCheck, "skip-free-space-check", false, "skips the check for storage free space prior to running the migrations")
+	flag.Parse()
+
+	logger := log.New(os.Stdout, "", 0)
+	cfg, err := config.GetConfig()
+	if err != nil {
+		logger.Printf("failed to get config: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+	cli, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		logger.Printf("failed to create kubernetes clientset: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+	if !skipFreeSpaceCheck {
+		checkFreeSpace(ctx, logger, cfg, cli, opts)
+	}
+
+	if err = migrate.Migrate(ctx, logger, cli, opts); err != nil {
+		fmt.Printf("%s\n", err.Error())
+		os.Exit(1)
+	}
+}
+
+func checkFreeSpace(ctx context.Context, logger *log.Logger, cfg *rest.Config, cli kubernetes.Interface, opts migrate.Options) {
```

Comment:
Why does this not return error? There's a lot of `os.Exit(1)` in it and that feels weird to me

Reply:
fixed.